### PR TITLE
#185, moving the use-in-operator to idiomatic and its relevant files

### PR DESCRIPTION
**Pull request by Parsifal-M**
Hey @anderseknert and Team! 👋 

Thought I'd try to lend a hand!

In terms of the `README`, I was not 100% sure how you envisioned it e.g.:

- Leave the readme where it is and update it saying it moved.
- Move the readme to the new location under `idiomatic` and update it saying it was previously under `style`

Let me know what you think is best here and I will update it 😄 

Not sure if I missed anything else! Please LMK and I'll get back to it!

This Fixes https://github.com/StyraInc/regal/issues/185

Thanks 🚀

### CODE REVIEW

File: docs/rules/idiomatic/use-in-operator.md [hunk 1]
```diff
@@ -39,12 +39,8 @@ This linter rule provides the following configuration options:
...
 ```yaml
 rules: 
-  style:
+  idiomatic:
     use-in-operator:
       # one of "error", "warning", "ignore"
       level: error
 ```
-
-## Related Resources
```
Comment (anderseknert) [inline docs/rules/idiomatic/use-in-operator.md:48]: I don't think we should remove this. While it's a style guide, this is still a good reference.

File: docs/rules/idiomatic/use-in-operator.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 **Summary**: Use `in` to check for membership
 
-**Category**: Style
```
Comment (anderseknert) [inline docs/rules/idiomatic/use-in-operator.md:5]: Considering backwards compatibility for this one, and apologies for not having done so in the issue 😅 But I think it's probably good to keep the` docs/rules/style/use-in-operator.md` file around for a while but have it contain only a link to the doc in the idiomatic category. That way, users of older versions would still find the docs and be able to update their config if needed. Does that make sense?

**Review by anderseknert — COMMENTED** [review]
Much appreciated, Peter! Just dropped a comment or two, but this is good 🙂

**Comment by anderseknert** [comment]
Also, don't forget to add "Fixes #185" to the commit message / PR description, in order to link the PR to the issue 🙂

File: docs/rules/idiomatic/use-in-operator.md (hunk 2 above)
Comment (Parsifal-M) [reply docs/rules/idiomatic/use-in-operator.md:5]: Ah yeah that makes sense now I think so basically two readme's but the one in `docs/rules/style/use-in-operator.md` just points to the "new" one (of course nicely) in `docs/rules/idiomatic/use-in-operator.md` am I following you on this? 😅

Comment (anderseknert) [reply docs/rules/idiomatic/use-in-operator.md:5]: Yep! We could just say something like "Rule moved to idiomatic category" and provide a link to its new location.

Comment (Parsifal-M) [reply docs/rules/idiomatic/use-in-operator.md]: Hey! 👋 

LMK what you think, happy to change it a bit more if needed 👍 . I added back the style reference guide and added a note to the old docs pointing to the new ones.

File: docs/rules/idiomatic/use-in-operator.md (hunk 1 above)
Comment (Parsifal-M) [reply docs/rules/idiomatic/use-in-operator.md]: Added it back 👍

**Review by anderseknert — APPROVED** [review]
Thanks Peter! 👏

**Pull request merged**